tests/lib/nested.sh

```diff
@@ -448,18 +523,24 @@ configure_cloud_init_nested_core_vm_uc20(){
     umount "$tmp"
 }
 
-get_nested_core_image_path(){
-    echo "$WORK_DIR/image/ubuntu-core.img"
-}
-
 force_stop_nested_vm(){
     systemctl stop nested-vm
 }
 
 start_nested_core_vm_unit(){
-    local IMAGE_FILE QEMU
-    IMAGE_FILE="$WORK_DIR/image/ubuntu-core-new.img"
+    local QEMU IMAGE_DIR IMAGE_NAME CURRENT_IMAGE
+    CURRENT_IMAGE="$WORK_DIR/image/ubuntu-core-current.img"
     QEMU=$(get_qemu_for_nested_vm)
+    IMAGE_DIR="$(get_image_dir)"
+    IMAGE_NAME="$(get_image_name core)"
+
+    # As core18 systems use to fail to start the assertion disk when using the
+    # snapshot feature, we copy the original image and use that copy to start
+    # the VM.
+    # Some tests however need to force stop and restart the VM with different
+    # options, so if that env var is set, we will reuse the existing file if it
+    # exists
+    cp "$IMAGE_DIR/$IMAGE_NAME" "$CURRENT_IMAGE"
```

Comment:
why was this moved into the start_nested_core_vm_unit function? the reason I introduced `start_nested_core_vm_unit` at all was so that we could avoid copying the image file and just start the vm systemd unit directly w/o modifying the image file, as there are cloud-init nested spread tests which rely on this functionality. I'm surprised those tests didn't fail when you did this, did you try running the nested cloud-init spread tests with this branch ?

Reply:
Thanks for the clarification, I already fixed it but I had to leave the cp inside the start_nested_core_vm_unit because in that function we calculate the name of the source image to copy. Previouly there was just 1 source image in the system, now and in the following PRs we could have more than 1 in order to reuse them. This is important to speed up in case we dont use kvm and we need to save time when running tests like refresh fundamentals.

Reply:
@anonymouse64 finally I moved back the cp to start_nested_core_vm and also added a comment to explain there the different scenarios for the current image.